### PR TITLE
Add validation of associated names. 

### LINKED ISSUE #391: Validate title note with type "associated name"
Create a new validator that validates that when title.note with type "associated name" has a value, it must match a contributor name.

When there is a mismatch, it should raise a validation error with the message "Missing data: Name associated with uniform title does not match any contributor."

refs https://github.com/sul-dlss/argo/issues/3375

### PR DESCRIPTION
**Pull request by justinlittman**
closes #391

**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔
* More validation
* Code re-use

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/cocina/generator/generator.rb [hunk 1]
```diff
@@ -122,10 +122,7 @@ def generate_for(schema)
 
       NO_CLEAN = [
         'checkable.rb',
-        'dro_rights_description_builder.rb',
```
Comment (justinlittman) [inline lib/cocina/generator/generator.rb:125]: These were moved into own module, so no longer needed here.

File: lib/cocina/models/builders/dro_rights_description_builder.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
```
Comment (justinlittman) [inline lib/cocina/models/builders/dro_rights_description_builder.rb:1]: This is just a move.

File: lib/cocina/models/builders/name_title_group_builder.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
```
Comment (justinlittman) [inline lib/cocina/models/builders/name_title_group_builder.rb:1]: This is copied from DSA.

File: lib/cocina/models/builders/rights_description_builder.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
```
Comment (justinlittman) [inline lib/cocina/models/builders/rights_description_builder.rb:1]: This is a move.

File: lib/cocina/models/builders/title_builder.rb [hunk 5]
```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
```
Comment (justinlittman) [inline lib/cocina/models/builders/title_builder.rb:1]: This is a move.

**Review by jcoyne — CHANGES_REQUESTED** [review]
This PR is very large and challenging to review. Can you do the moves in a separate PR?

**Comment by justinlittman** [comment]
This was tested on sdr-deploy.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**